### PR TITLE
OWA-34: Implement functionality to list addons in Add-on Manager Page

### PR DESCRIPTION
**Pull request by kingisaac95**
## JIRA TICKET NAME:
OWA-34 - [Implement functionality to list OWAs on Add-on Management page](https://issues.openmrs.org/browse/OWA-34)

### SUMMARY:
Users should see a table where all OWAs are listed with their details
The Add-on Manager should be able to list currently available OWAs. 
The OWA in the list should be clickable to redirect the user to OWA entry point

#### Screenshot
screen shot 2017-09-11 at 8 50 04 pm

### CODE REVIEW

**Comment by paulupendo** [comment]
@kingisaac95 Great job 😄

File: app/js/components/homePage/manageApps/manageApps.jsx [hunk 1]
```diff
@@ -8,21 +8,48 @@
...
+            <span className="glyphicon glyphicon-plus span-icon"/>  Add/Upgrade Add ons
           </button> 
           <button type="button" className="button-search-module">
-            <span className="glyphicon glyphicon-search span-icon search-icon"></span>Search Add ons
+            <span className="glyphicon glyphicon-search span-icon search-icon"/>Search Add ons
           </button> 
         </div>
-        <div className="manage-app-table col-sm-10">
-          <table className="table table-bordered">
+        <div className="manage-app-table col-sm-12">
```
Comment (jomadoye) [inline app/js/components/homePage/manageApps/manageApps.jsx]: @kingisaac95 Can we move the `listing Addon table` to a new component, e.g `ListAddon.jsx`

**Comment by jomadoye** [comment]
@kingisaac95 Great job 😄.

File: app/js/components/homePage/manageApps/manageApps.jsx (hunk 1 above)
Comment (kingisaac95) [reply app/js/components/homePage/manageApps/manageApps.jsx]: I've done that. I have `AddonList.jsx` that renders the table body.
Thanks @andela-jomadoye

**Comment by kingisaac95** [comment]
Thanks guys

File: app/js/components/homePage/manageApps/manageApps.jsx (hunk 1 above)
Comment (kingisaac95) [reply app/js/components/homePage/manageApps/manageApps.jsx]: I've done that. I have AddonList.jsx that renders the table body.
Thanks @andela-jomadoye

**Comment by paulupendo** [comment]
@kingisaac95 Would we please squash commits. Before the merge.

**Comment by kingisaac95** [comment]
@paulupendo Thanks, I just noticed that my commit history got messed up. I've tried to fix that.

**Comment by kingisaac95** [comment]
@dkayiwa Please I have fixed the commit issue. My commit history got messed up that why multiple commits showed up. Apologies for that, but you can review it now

**Comment by dkayiwa** [comment]
@kingisaac95 did you see my last comment on the ticket?

**Comment by kingisaac95** [comment]
I tried to reply to the comment but I couldn't add a comment there. Can we do the call tomorrow morning @dkayiwa?

**Comment by dkayiwa** [comment]
It is all up to you.

**Comment by kingisaac95** [comment]
If you're free now, we can still do it @dkayiwa - https://hangouts.google.com/hangouts/_/andela.com/syncwithdaniel?authuser=0

**Comment by dkayiwa** [comment]
already there.

**Comment by kingisaac95** [comment]
Thanks @dkayiwa. I have fixed the commit history issue and so the files are clean now with only 6 file changes. Also I have written tests for the new component I created which is included in this PR. I have also attached the screenshot on the PR description and the JIRA page.

File: tests/homepage/addonList.test.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        "version": "1.0.0",
+        "name": "CohortBuilder",
+        "description": "modern refresh of the OpenMRS Cohort Builder tool",
+        "icons": {
+            "16": null,
+            "48": "/img/omrs-button.png",
+            "128": null
+        },
+        "developer": {
+            "url": "https://github.com/openmrs/openmrs-owa-cohortbuilder",
```
Comment (dkayiwa) [inline tests/homepage/addonList.test.js:18]: What is cohortbuilder doing here? ☺️

File: tests/homepage/addonList.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            "name": "andela-odaniel, wanjikum, andela-jomadoye, andela-pupendo, kingisaac95",
```
Comment (dkayiwa) [inline tests/homepage/addonList.test.js:19]: Have all these devs contributed to this?

File: tests/homepage/addonList.test.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { expect } from 'chai';
+import { mount, shallow } from 'enzyme';
+import  { AddonList }  from '../../app/js/components/homePage/manageApps/addonList';
+
+
+describe('<AddonList />', () => {
+    const appList = [{
+        "version": "1.0.0",
+        "name": "CohortBuilder",
```
Comment (dkayiwa) [inline tests/homepage/addonList.test.js:10]: Cohort what????

File: tests/homepage/addonList.test.js (hunk 2 above)
Comment (kingisaac95) [reply tests/homepage/addonList.test.js:18]: It's a test object I'm passing as props so I can test the behavior of the `addonList.jsx` component

File: tests/homepage/addonList.test.js (hunk 3 above)
Comment (kingisaac95) [reply tests/homepage/addonList.test.js:19]: No, it's for the Cohort builder. To test that the `addonList` compnent renders what comes from the API, I got an object from the API response and I'm using it to test it

File: tests/homepage/addonList.test.js (hunk 4 above)
Comment (kingisaac95) [reply tests/homepage/addonList.test.js]: :) Ditto

Comment (kingisaac95) [reply tests/homepage/addonList.test.js:10]: 😄 Ditto

File: tests/homepage/addonList.test.js (hunk 2 above)
Comment (dkayiwa) [reply tests/homepage/addonList.test.js:18]: Are you testing the cohort builder or addonmanager?

**Comment by wanjikum** [comment]
Good work @kingisaac95!

**Comment by kingisaac95** [comment]
@dkayiwa Please is there still anything I should fix?

**Comment by dkayiwa** [comment]
@kingisaac95 there is still a comment you have not responded to, on this very pull request.

File: tests/homepage/addonList.test.js (hunk 2 above)
Comment (kingisaac95) [reply tests/homepage/addonList.test.js:18]: @dkayiwa I'm testing the `addonList` component of Addon manager. The object I used there was only to have values passed down to the component during testing. It is for testing purpose alone and so is not part of the component code

**Pull request merged**